perception/autoware_image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp

```diff
@@ -122,8 +122,11 @@ void SegmentPointCloudFusionNode::fuse_on_single_image(
       *reinterpret_cast<float *>(&transformed_cloud.data[global_offset + y_offset]);
     float transformed_z =
       *reinterpret_cast<float *>(&transformed_cloud.data[global_offset + z_offset]);
-    // skip filtering pointcloud behind the camera or too far from camera
-    if (transformed_z <= 0.0 || transformed_z > filter_distance_threshold_) {
+    // skip pointcloud out of FOV
+    if (det2d_status.camera_projector_ptr->isOutsideHorizontalView(transformed_x, transformed_z)) {
+      continue;
+    }
+    if (transformed_z > filter_distance_threshold_) {
```

Comment:
[Q] Do you have any specific reason to separate these conditions?

Reply:
@ktro2828 

> [Q] Do you have any specific reason to separate these conditions?

The separation is just for beter readabiltiy. 
Note that `transformed_z <= 0.0` can only ignore pointcloud behind the camera, by adding `isOutsideHorizontalView`, more pointcloud (a part of pointcloud ahead) also can be skipped.